modules/collect/src/main/java/com/opengamma/strata/collect/NumberFormatter.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright (C) 2018 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.collect;
+
+import java.math.RoundingMode;
+import java.text.DecimalFormat;
+import java.text.DecimalFormatSymbols;
+import java.text.NumberFormat;
+import java.text.ParseException;
+import java.util.Locale;
+
+/**
+ * Provides the ability to parse and format numbers.
+ * <p>
+ * This exists as an alternative to {@link NumberFormat} and {@link DecimalFormat}
+ * which are not thread-safe.
+ * <p>
+ * Instances of this class are immutable and thread-safe.
+ */
+public final class NumberFormatter {
+
+  /**
+   * The underlying format.
+   */
+  private final ThreadLocal<NumberFormat> underlying;
+
+  /**
+   * Restricted constructor.
+   */
+  private NumberFormatter(NumberFormat format) {
+    ArgChecker.notNull(format, "format");
+    format.setParseIntegerOnly(false);
+    this.underlying = ThreadLocal.withInitial(() -> (NumberFormat) format.clone());
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Obtains a standard formatter configured by grouping and decimal places.
+   * <p>
+   * The formatter will have the specified number of decimal places.
+   * The integer part will be grouped if the flag is set.
+   * The decimal part will never be grouped or truncated.
+   * The implementation uses English locale data, which uses commas as a separator and a decimal point (dot).
+   * Numbers will be rounded using {@link RoundingMode#HALF_EVEN}
```

Comment:
Why `HALF_EVEN` and not `HALF_UP`? `HALF_UP` is the "standard" rounding mode everyone learns at school. From the Javadocs for `HALF_UP`

>Note that this is the rounding mode commonly taught at school.

Reply:
It is what `DecimalFormat` uses.

Reply:
So it does. That surprises me.